### PR TITLE
HPCC-29963 Correct XSD type conversion for TpQueryType in WsTopology Service

### PR DESCRIPTION
**Pull request by kenrowland**
Added exception for type TpQueryType in XSLT to ensure generated XSD has proper type of xsd:string

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by asselitx — APPROVED** [review]

File: esp/xslt/esp_service_xml2xsd.xslt [hunk 1]
```diff
@@ -153,6 +153,8 @@
                     </xsl:when>
                     <xsl:when test="@type='int64'">xsd:<xsl:value-of select="'long'"/>
                     </xsl:when>
+                    <xsl:when test="@type='TpQueryType'">xsd:<xsl:value-of select="'string'"/>
```
Comment (afishbeck) [inline esp/xslt/esp_service_xml2xsd.xslt:156]: @kenrowland This is actually covering up a bug in ws_topology.ecm.  TpQueryType was meant to be defined as an enumerated string.  But it was incorrectly defined and incorrectly used.

You could change the definition of TpQueryType from ESPstruct to ESPenum and then make use of it, BUT that may cause backward compatibility issues.  
The alternative for maximum backward compatibility would be just to change it to a string.

@wangkx when you were changing strings to enums how did we handle backward compatability?

Anyway I wouldn't make this change, one way or the other I would fix the bad item in the ecm.

Comment (wangkx) [reply esp/xslt/esp_service_xml2xsd.xslt:156]: In ecm file, add a new variable and 'depr_ver' the old one. If the variables are in a request, make sure that the new variable is read first. If not found, read the old one for backward compatibility.

Comment (afishbeck) [reply esp/xslt/esp_service_xml2xsd.xslt:156]: @kenrowland for now I would just declare it as a string but add a comment about the two expected values (from the struct definition).  It's no worse or less informative than what is there now.

**Review by afishbeck — CHANGES_REQUESTED** [review]
Shouldn't hard code this change, rather fix the errors in the ecm.

**Review by afishbeck — APPROVED** [review]

**Review by wangkx — APPROVED** [review]

**Comment by afishbeck** [comment]
@kenrowland the PR and commit titles and comments are no longer correct.

**Comment by GordonSmith** [comment]
@kenrowland can you squash and double check the commit message?

**Comment by kenrowland** [comment]
@GordonSmith Please merge

**Pull request merged**